test/unit/org/apache/cassandra/hints/HintsServiceTest.java

```diff
@@ -149,6 +151,44 @@ public void testHintsDroppedForUnknownHost()
         );
     }
 
+    @Test
+    public void testCountingCorruptedHints()
+    {
+        // pause the scheduled dispatch before writing hints
+        HintsService.instance.pauseDispatch();
+        try
+        {
+            long hintsOnDisk0 = HintsServiceMetrics.hintsOnDisk.getCount();
+            long corruptedHintsOnDisk0 = HintsServiceMetrics.corruptedHintsOnDisk.getCount();
+
+            // write 100 hints on disk for host that is not part of cluster
```

Comment:
This makes them corrupt?

Reply:
no, this is just a stupid test which check that the counter increments by the number of hints in the file and then decrements when we delete hints